modules/heka/msg_interpolate.lua

```diff
@@ -0,0 +1,99 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+## Simple Templating to Transform Message Fields into Strings
+
+### API
+
+#### interpolate(value, secs)
+
+Interpolates values from the currently processed message into the provided
+string value. A `%{}` enclosed field name will be replaced by the field value
+from the current message. All message header fields are supported ("Uuid",
+"Timestamp", "Type", "Logger", "Severity", "Payload", "EnvVersion", "Pid",
+"Hostname"). Any other values will be checked against the defined dynamic
+message fields. If no field matches, then a
+[C strftime](http://man7.org/linux/man-pages/man3/strftime.3.html) (*nix)
+or
+[C89 strftime](http://msdn.microsoft.com/en-us/library/fe06s4ak.aspx) (Windows)
+time substitution will be attempted. The time used for time substitution will be
+the seconds-from-epoch timestamp passed in as the `secs` argument, if provided.
+If `secs` is nil, local system time is used. Note that the message timestamp is
+*not* automatically used; if you want to use the message timestamp for time
+substitutions, then you need to manually extract it and convert it from
+nanoseconds to seconds (i.e. divide by 1e9).
+
+*Arguments*
+* value (string) - String into which message values should be interpolated.
+* secs (number or nil) - Timestamp (in seconds since epoch) to use for time
+  substitutions. If nil, system time will be used.
+
+*Return*
+* string - Original string value with any interpolated message values.
+--]]
+
+local date          = require "os".date
+local floor         = require "math".floor
+local string        = require "string"
+local read_message  = read_message
+local tostring      = tostring
+local type          = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+local function interpolate_match(header_fields, match, secs)
+    -- First see if it's a message header.
+    local fp = header_fields[match]
+    if fp then return fp() end
+
+    -- Second check for a dynamic field.
+    local fname = string.format("Fields[%s]", match)
+    local fval = read_message(fname)
+    if type(fval) == "boolean" then
+        return tostring(fval)
```

Comment:
so they return something to substitute

On Mon, Apr 4, 2016 at 10:53 AM, Mark Reid <notifications@github.com> wrote:

> In modules/heka/msg_interpolate.lua
> <https://github.com/mozilla-services/lua_sandbox/pull/127#discussion_r58417779>
> :
>
> > +local tostring      = tostring
> > +local type          = type
> > +
> > +local M = {}
> > +setfenv(1, M) -- Remove external access to contain everything in the module.
> > +
> > +local function interpolate_match(header_fields, match, secs)
> > +    -- First see if it's a message header.
> > +    local fp = header_fields[match]
> > +    if fp then return fp() end
> > +
> > +    -- Second check for a dynamic field.
> > +    local fname = string.format("Fields[%s]", match)
> > +    local fval = read_message(fname)
> > +    if type(fval) == "boolean" then
> > +        return tostring(fval)
>
> why do boolean fields get tostring'd?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla-services/lua_sandbox/pull/127/files/fc4f4f68322970337fd87bb3258f5e9d8108c07b#r58417779>
>